File: src/AnalyticsService/Service/Commands/InsertBatchStatCommand.cs
using AnalyticsService.Database.Model.Domain;
using MediatR;

namespace AnalyticsService.Service.Commands;

/// <summary>
/// Command for inserting a new stat to the database.
/// </summary>
internal sealed record InsertBatchStatCommand(
    DateTime StartDate,
    DateTime EndDate,
    int DocsNumber,
    BatchStatus Status,
    Guid WorkflowId
) : IRequest<bool>;

File: src/AnalyticsService/Service/Commands/InsertBatchStatCommand.fs
namespace AnalyticsService.Service.Commands

open System
open MediatR

/// Command for inserting a new stat to the database.
[<Sealed>]
type internal InsertBatchStatCommand(
    startDate: DateTimeOffset,
    endDate: DateTimeOffset,
    docsNumber: int,
    status: int,
    workflowId: Guid
) =
    interface IRequest<bool>
    member this.StartDate = startDate
    member this.EndDate = endDate
    member this.DocsNumber = docsNumber
    member this.Status = status
    member this.WorkflowId = workflowId